### PR TITLE
Fix spec feast validator for case of using expression in entity extraction

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:

Fix standard transformer spec validation for case when expression is used within feast entity extraction. E.g.
```
transformerConfig:
  preprocess:
    inputs:
      - variables:
          - name: customer_level
            literal:
              intValue: 10
      - feast:
          - tableName: customer_feature_table
            project: default
            entities:
              - name: customer_level
                valueType: INT32
                expression: customer_level
            features:
              - name: customer_feature_1
                valueType: INT64
                defaultValue: '0'
...
```
Currently, the above spec will cause validation error mentioning that `customer_level` expression is not defined even though it's declared in the variable block.
The issue is due to validator not reusing symbol registry which should contains the declared variables.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/pkg/transformer/pipeline/compiler_test.go [hunk 1]
```diff
@@ -258,6 +258,77 @@ func TestCompiler_Compile(t *testing.T) {
...
+					CacheEnabled: true,
+				},
+				cacheOptions: &cache.Options{
+					SizeInMB: 100,
+				},
+				logger: logger,
+			},
+			specYamlFilePath: "./testdata/valid_feast_expression.yaml",
+			want: want{
+				expressions: []string{
```
Comment (ariefrahmansyah) [inline api/pkg/transformer/pipeline/compiler_test.go:311]: can you add test for UDF expresion, such as Now(), or Geohash()?

File: api/pkg/transformer/pipeline/testdata/valid_feast_expression.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                defaultValue: '0'
+          - tableName: customer_feature_table
+            project: default
+            entities:
+              - name: customer_id
+                valueType: STRING
+                expression: customer_id
+              - name: customer_level
+                valueType: INT32
+                expression: customer_level
```
Comment (ariefrahmansyah) [inline api/pkg/transformer/pipeline/testdata/valid_feast_expression.yaml:38]: Can you also add UDF expression here? Such as Now() or Geohash()

Comment (pradithya) [reply api/pkg/transformer/pipeline/testdata/valid_feast_expression.yaml:38]: Added

File: api/pkg/transformer/pipeline/compiler_test.go (hunk 1 above)
Comment (pradithya) [reply api/pkg/transformer/pipeline/compiler_test.go:311]: Added

**Review by ariefrahmansyah — APPROVED** [review]
lgtm

**Pull request merged**